webui/react/src/routes/utils.ts

```diff
@@ -4,16 +4,25 @@ import { MouseEvent, MouseEventHandler } from 'react';
 import history from 'routes/history';
 import { Command, CommandType } from 'types';
 import { clone } from 'utils/data';
+import { globalStorage } from 'utils/storage';
 
 import routes from './routes';
+import { RouteConfig } from './types';
 
+// serverAddress returns determined cluster (master) address.
 export const serverAddress = (avoidDevProxy = false, path = ''): string => {
   if (!!path && !isAbsolutePath(path)) return path;
 
-  const address = avoidDevProxy && process.env.IS_DEV
-    ? 'http://localhost:8080'
-    : `${window.location.protocol}//${window.location.host}`;
-  return address + path;
+  // Prioritize dynamically set address.
+  const customServer = globalStorage.get('server-address') || process.env.SERVER_ADDRESS;
+
+  // FIXME trailing slashes
```

Comment:
**chore:** does this just need something like this? `path.replace(/\/\s*$/, '')`

Reply:
or better wherever we set the inputs we make sure it's correct.

Reply:
as in we could move the validation and clean up closer to the user input :thinking:

Reply:
d5c8dae26657cfb3eef1e1cf0e18687c36be6795